File: backend/pkg/response/response.go
package response

import "github.com/gofiber/fiber/v3"

type Response struct {
	Success bool   `json:"success"`
	Message string `json:"message"`
	Data    any    `json:"data,omitempty"`
}

func JSON(c fiber.Ctx, status int, success bool, message string, data any) error {
	return c.Status(status).JSON(Response{
		Success: success,
		Message: message,
		Data:    data,
	})
}

File: backend/src/shared/utils/response.ts
import { Response } from 'express';

export interface ApiResponse<T = unknown> {
  success: boolean;
  message: string;
  data?: T;
}

export function sendJson<T>(
  res: Response,
  status: number,
  success: boolean,
  message: string,
  data?: T,
): Response {
  const body: ApiResponse<T> = { success, message };
  if (data !== undefined) body.data = data;
  return res.status(status).json(body);
}
